lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -583,6 +615,38 @@ public function setBackupExisting($bool)
     }
 
     /**
+     * Set whether or not to add PHP strict types.
+     *
+     * @param bool $bool
+     *
+     * @return void
+     */
+    public function setStrictTypes($bool)
+    {
+        if ($bool && version_compare(PHP_VERSION, '7.0.0', '<')) {
```

Comment:
Are there any use-cases where people might need to *generate* the source from a different machine (possibly with a different PHP version) than where they plan to use it?